src/img_to_text_gen.py

```diff
@@ -0,0 +1,57 @@
+import torch
+from transformers import VisionEncoderDecoderModel, ViTFeatureExtractor, AutoTokenizer
+
+itt_model = None
+itt_feature_extractor = None
+itt_tokenizer = None
+itt_device = None
+itt_inited = False
+
+def init_itt():
+    global itt_model
+    global itt_feature_extractor
+    global itt_tokenizer
+    global itt_device
+    global itt_inited
+
+    if itt_inited:
+        return itt_inited
+
+    try:
+        itt_model = VisionEncoderDecoderModel.from_pretrained("nlpconnect/vit-gpt2-image-captioning")
+        itt_feature_extractor = ViTFeatureExtractor.from_pretrained("nlpconnect/vit-gpt2-image-captioning")
+        itt_tokenizer = AutoTokenizer.from_pretrained("nlpconnect/vit-gpt2-image-captioning")
+
+        itt_device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        itt_model.to(itt_device)
+    except:
+        itt_inited = False
+        return False
+
+    itt_inited = True
+    return itt_inited
+
+def gen_itt(image):
+    if not itt_inited:
+        return None
+    if image is None:
+        return None
+
+    max_length = 16
+    num_beams = 4

```

Comment:
These are very sensitive values for text generation. Let's add them to an interface, so someone may play with them.

Reply:
You can use sliders from streamlit, for example

Reply:
These parameters have almost no effect on the result. And it takes a long time to regenerate a description. Because of it I do not want to give an ability to change values